### DISCUSSION TITLE
Big publication stuck - Upgrading new DITAOT from DITA-OT version 1.8

### DISCUSSION THREAD

**Discussion by ganapathy1** [dita-ot/dita-ot#4536] [Q&A] [answered]
Hi All,

Big publication stuck - Upgrading new DITAOT from DITA-OT version 1.8.

We recently upgraded from DITA-OT 1.8 to version 4.2.3. We copied all the old custom plugins into the new version without changing any XSLT code, and the upgrade was successful. Small publications are working fine, but larger publications are getting stuck. Please share any ideas on how to resolve this issue."

**Comment by raducoravu** [comment]
Could you give more details? What do you mean by “stuck”? Maybe you could run the dita cmd line with the -v param and post the last 10-20 lines of console output.

**Reply by ganapathy1** [reply]
I am facing an issue with Oxygen. After creating the stage1.xml file, the process of converting it into a .fo file is taking too long. The .fo file printing line by line and it is taking almost an entire day to generate the PDF. only for big ditamap file.

**Reply by jelovirt** [reply]
What's the file size of stage1.xml and *.fo file? Does it ever generate the final PDF?

What happens if you publish the same map to PDF without any custom plug-ins?

**Comment by raducoravu** [comment]
@ganapathy1 I do not understand what you mean by "The .fo file printing line by line", the DITA OT console output should not print the contents of the FO file unless you have maybe added extra xsl:messages in the stylesheets. If so, maybe you can remove your xsl:messages and see if the publishing is faster.

You could also try to increase the amount of memory allocated to the DITA OT process:
https://www.oxygenxml.com/doc/versions/26.1/ug-editor/topics/dita-map-edit-advanced.html#dita-map-edit-advanced__jvm-arguments

**Reply by ganapathy1** [reply, marked as answer]
Thank you for your reply.
We migrated our customization code from DITA-OT version 1.8 and copied all the customization plugins to the new version, DITA-OT 4.2.3. While a small input file generates the PDF output correctly, larger input files are not generating the PDF and are getting stuck in the stage2.fo processing step. After few analaysis in the code, I have noticed that common.xsl and table.xsl are the main reasons for the slowness issue in our customization folder for PDF generation. I copied these files from the default plugin org.dita.pdf2, and now the PDF is generated within 3 minutes for large publications. I am workng in table and common xslt part to fix this issue.
